### PR TITLE
refactor: bump ruff's minimum python version to 3.8 and remove remaining python 3.7 code

### PR DESCRIPTION
**Pull request by bellini666**

### CODE REVIEW

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/), so make sure to pick the appropriate type. If in doubt feel free to ask :)


Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to Thiago Bellini Ribeiro for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

File: strawberry/utils/cached_property.py [hunk 1]
```diff
@@ -1,11 +1,6 @@
-import sys
+from functools import cached_property
```
[the remark this answers is not in the document]
Comment (patrick91) [reply strawberry/utils/cached_property.py:1]: yup!

[the remark this answers is not in the document]
Comment (bellini666) [reply strawberry/utils/cached_property.py:1]: Done!

**Comment by bellini666** [comment]
@patrick91 the tests are fine other than the lack of a `RELEASE.md` file.

Do you think we need to make a release for this change? It is mostly internal refactoring stuff

**Review by patrick91 — APPROVED** [review]
Looks good, I've added a release file, and removed `backports.cached-property` from the deps 😊

**Comment by bellini666** [comment]
> Looks good, I've added a release file, and removed `backports.cached-property` from the deps blush

Awesome! Forgot to do that :)

**Pull request merged**